README.md
<!-- README.md is generated from README.Rmd. Please edit that file -->
Damn Fine Color Palettes
========================

![](figure/damnfine.gif)

> That's a damn fine plot, Diane - Special Agent Dale Cooper (probably)

The wesanderson package claimed that "Short of adding an owl and dressing up your plot in a bowler hat, here's the most indie thing you can do to one." So I took them up on that challenge and made the damnfinecolorpalette where the owls are not what they seem. First round of palettes derived from the video [Color By Numbers: Twin Peaks](https://vimeo.com/218231355), the instagram [colorpalatte.cinema](https://www.instagram.com/colorpalette.cinema/), and the use for a color picker extension and screenshots. Concept and code based on the wonderful wesanderson palette generator.

Installation
------------

**Development version**

``` r
devtools::install_github("fbenamy/damnfinecolorpalette")
```

Usage
-----

``` r
library(damnfinecolorpalette)

# See all palettes
names(damnfinecolors)
#>  [1] "RedRoom1"     "RedRoom2"     "BeckyBriggs"  "LumberMill"  
#>  [5] "PhoneBooth"   "PeteMartell"  "OneEyedJacks" "RRDiner"     
#>  [9] "Dougie"       "AudreyHorne"  "LogLady"      "SpecialAgent"
```

Palettes
--------

### The Red Room

``` r
damnfine("RedRoom1")
```

![](figure/RedRoom-1.png)

``` r
damnfine("RedRoom2")
```

![](figure/RedRoom-2.png)

### Becky Briggs

``` r
damnfine("BeckyBriggs")
```

![](figure/unnamed-chunk-3-1.png)

### The Lumber Mill

``` r
damnfine("LumberMill")
```

![](figure/rushmore-1.png)

### Phone Booth

``` r
damnfine("PhoneBooth")
```

![](figure/unnamed-chunk-4-1.png)

### PeteMartell

``` r
damnfine("PeteMartell")
```

![](figure/unnamed-chunk-5-1.png)

### One Eyed Jacks

``` r
damnfine("OneEyedJacks")
```

![](figure/OneEyedJacks-1.png)

### RR Diner

``` r
damnfine("RRDiner")
```

![](figure/RRDiner-1.png)

### Dougie

``` r
damnfine("Dougie")
```

![](figure/Dougie-1.png)

### Audrey Horne

``` r
damnfine("AudreyHorne")
```

![](figure/Audrey%20Horne-1.png)

``` r
library("ggplot2")
ggplot(mtcars, aes(factor(cyl), fill=factor(vs))) +  geom_bar(color = "black") +
  scale_fill_manual(values = damnfine("AudreyHorne"))
```

![](figure/ggplot1-1.png)

### The Log Lady

``` r
damnfine("LogLady")
```

![](figure/Log%20Lady-1.png)

``` r
pal <- damnfine("LogLady", 21, type = "continuous")
image(volcano, col = pal)
```

![](figure/volcano-1.png)

``` r
pal <- damnfine("LogLady", 100, type = "continuous")
# heatmap is a local dataset
ggplot(heatmap, aes(x = X2, y = X1, fill = value)) +
  geom_tile() + 
  scale_fill_gradientn(colours = pal) + 
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) + 
  coord_equal() 
```

![](figure/log_lady_heatmap-1.png)

### Special Agent

``` r
damnfine("SpecialAgent")
```

![](figure/specialagent-1.png)


NEWS
Version 0.1
=============
* “One day my log will have something to say about this."
This is my first attempt at modifying an R package, thank you to the creators of wespalette: Karthik Ram and Hadley Wickham


R/colors.R
#' Complete list of palettes
#'
#' Use \code{\link{damnfine}} to construct palettes of desired length.
#'
#' @export
damnfinecolors <- list(
  RedRoom1 = c("#1C1003", "#74574E", "#3C2802", "#785828", "#C77928", "#5D110C", "#0E3442", "#436176",
              "#5A3630", "#6B5A67", "#9C878E"),
  RedRoom2 = c("#C2DDC7", "#5B1113", "#7A1317", "#62372D", "#110706",  "#A51A25", "#423330", "#E1B692", "#8D644E", "#D62639"),
  BeckyBriggs = c("#1E0002", "#490103", "#820308", "#AB070F", "#F06E73", "#FBB59E", "#F0A489", "#B9A270", "#C8923D", "#AF7443"),
  LumberMill = c("#492618", "#B3592E", "#747B7D", "#71473B", "#C97D59"),
  PhoneBooth = c("#1B1210", "#A43A15", "#FF6C29", "#F97453", "#F1AB9B", "#C1845E", "#FEF8D8"),
  PeteMartell = c("#562110", "#923616", "#5B4E2F", "#E68429", "#FBAD4D"),
  OneEyedJacks = c("#522B23", "#7E1F0E", "#D83418", "#E66A4E", "#FFCFA4"),
  RRDiner = c( "#f4c33a",'#8E1F1F',"#844526", "#84d6cc", "#EBDCE0"),
  Dougie = c("#1C1210", "#666B19", "#FAD0CB", "#25210E", "#A18068", "#593C27"),
  AudreyHorne = c( "#492E25", "#E66F5D", "#D24E3B", "#EE3A2F", "#D08B8E"),
  LogLady = c("#dd0d2c", "#E2B8A3", "#749092", "#72502E" , "#310A15"),
  SpecialAgent = c("#A41822", "#FFFFFF", "#16080B", "#555051")
)

#' A Twin Peaks palette generator
#'
#' These are color palettes from the television show Twin Peaks and the movie
#' Fire Walk with Me.
#'
#' @param n Number of colors desired. RedRoom1, LumberMill, PhoneBooth,
#' PeteMartell and OneEyedJacks color schemes are from the video
#'  \href{https://vimeo.com/218231355}{Color by Numbers: Twin Peaks}.. BeckyBriggs and
#' RedRoom2 are from the instagram
#' \href{https://www.instagram.com/colorpalette.cinema/}{colorpalette.cinema}
#'   If omitted, uses all colours.
#' @param name Name of desired palette. Choices are:
#'   \code{RedRoom}, \code{LumberMill},  \code{PhoneBooth},
#'   \code{PeteMartell}, \code{OneEyedJacks}
#' @param type Either "continuous" or "discrete". Use continuous if you want
#'   to automatically interpolate between colours.
#'   @importFrom graphics rgb rect par image text
#' @return A vector of colours.
#' @export
#' @keywords colors
#' @examples
#' damnfine("RedRoom")
#' damnfine("OneEyedJacks")
#' damnfine("LumberMill")
#' damnfine("LumberMill", 3)
#'
#' # If you need more colours than normally found in a palette, you
#' # can use a continuous palette to interpolate between existing
#' # colours
#' pal <- damnfine(21, name = "PhoneBooth", type = "continuous")
#' image(volcano, col = pal)
damnfine <- function(name, n, type = c("discrete", "continuous")) {
  type <- match.arg(type)

  pal <- damnfinecolors[[name]]
  if (is.null(pal))
    stop("Palette not found.")

  if (missing(n)) {
    n <- length(pal)
  }

  if (type == "discrete" && n > length(pal)) {
    stop("Number of requested colors greater than what palette can offer")
  }

  out <- switch(type,
    continuous = grDevices::colorRampPalette(pal)(n),
    discrete = pal[1:n]
  )
  structure(out, class = "palette", name = name)
}

#' @export
#' @importFrom graphics rect par image text
#' @importFrom grDevices rgb
print.palette <- function(x, ...) {
  n <- length(x)
  old <- par(mar = c(0.5, 0.5, 0.5, 0.5))
  on.exit(par(old))

  image(1:n, 1, as.matrix(1:n), col = x,
    ylab = "", xaxt = "n", yaxt = "n", bty = "n")

  rect(0, 0.9, n + 1, 1.1, col = rgb(1, 1, 1, 0.8), border = NA)
  text((n + 1) / 2, 1, labels = attr(x, "name"), cex = 1, family = "serif")
}

#' heatmap
#'
#' A heatmap example
"heatmap"
